kimchi/src/tests/framework.rs

```diff
@@ -134,4 +139,51 @@ impl TestFramework {
         verify::<Affine, BaseSponge, ScalarSponge>(&group_map, &verifier_index, &proof).unwrap();
         println!("- time to verify: {}ms", start.elapsed().as_millis());
     }
+
+    /// Create and verify proof with deserialization
+    pub(crate) fn run_test_serialization(
+        gates: Vec<CircuitGate<Fp>>,
+        witness: [Vec<Fp>; COLUMNS],
+        public: &[Fp],
+    ) {
+        // create the index
+        let start = Instant::now();
+        let index = new_index_for_test(gates, public.len());
+        let verifier_index = index.verifier_index();
+        println!("- time to create index: {:?}s", start.elapsed().as_secs());
+
+        let verifier_index_serialize =
+            serde_json::to_string(&verifier_index).expect("couldn't serialize index");
+
+        // verify the circuit satisfiability by the computed witness
+        index.cs.verify(&witness, public).unwrap();
+
+        // add the proof to the batch
+        let start = Instant::now();
+        let group_map = <Affine as CommitmentCurve>::Map::setup();
+        let proof =
+            ProverProof::create::<BaseSponge, ScalarSponge>(&group_map, witness, &index).unwrap();
+        println!("- time to create proof: {:?}s", start.elapsed().as_secs());
+
+        // deserialize the verifier index
+        let mut verifier_index_deserialize: VerifierIndex<GroupAffine<VestaParameters>> =
+            serde_json::from_str(&verifier_index_serialize).unwrap();
+
+        // add srs with lagrange bases
+        let mut srs = SRS::<GroupAffine<VestaParameters>>::create(verifier_index.max_poly_size);
+        srs.add_lagrange_basis(verifier_index.domain);
+        verifier_index_deserialize.srs = Arc::new(srs);
+        verifier_index_deserialize.fq_sponge_params = oracle::pasta::fq_kimchi::params();
+        verifier_index_deserialize.fr_sponge_params = oracle::pasta::fp_kimchi::params();
+        verifier_index_deserialize.zkpm = zk_polynomial(verifier_index_deserialize.domain);
+        verifier_index_deserialize.powers_of_alpha = index.powers_of_alpha;
+        verifier_index_deserialize.linearization = index.linearization;
+        verifier_index_deserialize.w = zk_w3(verifier_index_deserialize.domain);
+
+        // verify the proof
+        let start = Instant::now();
+        verify::<Affine, BaseSponge, ScalarSponge>(&group_map, &verifier_index_deserialize, &proof)
+            .unwrap();
+        println!("- time to verify: {}ms", start.elapsed().as_millis());
+    }
```

Comment:
don't think this should be in `framework.rs` btw, as this looks more like a test than boilerplate code that is useful for creating other tests

Reply:
shall i extract this part out and place it in the newly created `serialization.rs` ?

Reply:
yeah I think that's a good idea